### ISSUE TITLE
Request: Set / boolean icons (Union, Intersect, Subtract, Exclude)

### ISSUE THREAD

**Issue by graup** [lucide-icons/lucide#2413]
### Icon name

Set{Union, Intersect, Subtract, Extract}

### Use cases

These can be needed in design contexts as well as when operating on data sets.

### Design ideas

In Figma: https://help.figma.com/hc/en-us/articles/360039957534-Boolean-operations

In Sketch: https://www.invisionapp.com/inside-design/how-to-make-custom-icons-in-sketch-using-boolean-operations/

It's a bit difficult to represent in Lucide's design language as we don't use any fills. But should be possible.

### Checklist

- [X] I have searched if someone has submitted a similar issue before and there weren't any. (Please make sure to also search closed issues, as this issue might already have been resolved.)
- [X] I have searched existing icons to make sure it does not already exist and I didn't find any.
- [X] I am not requesting a brand logo and the art is not protected by copyright.
- [X] I am not requesting an icon that includes religious, political imagery or hate symbols.
- [X] I have provided appropriate use cases for the icon(s) requested.

**Comment by karsa-mistmere** [comment]
Also see https://github.com/lucide-icons/lucide/pull/1151#issuecomment-1550963332

**Comment by graup** [comment]
> Also see [#1151 (comment)](https://github.com/lucide-icons/lucide/pull/1151#issuecomment-1550963332)

Oh yes, thanks! There are already some nice suggestions in that thread

**Comment by graup** [comment]
Here is my take on it. Could try to get this into a PR if this direction seems good

![Screenshot 2024-08-28 at 17 23 22](https://github.com/user-attachments/assets/c43ed380-f979-4da6-b1b4-a28cfa000f9e)

**Comment by jamiemlaw** [comment]
Potential alternatives:
![image](https://github.com/user-attachments/assets/549fb9ef-7f8c-4d92-896a-3fb7b5965282)

The top row most closely matches @danielbayley's designs, with the dashes being the same length as those elsewhere in Lucide.

The second row has, I think, nicer intersect and difference icons, but they are all off-centre.

The third row has longer dashes but I think is the best looking on average.

**Comment by graup** [comment]
I would agree the third row looks best. The first row looks a bit off. Especially the "intersect" middle being so small is not good I think, as the middle is the most meaningful part of that icon.
